telegram/ext/_fsm/states.py

```diff
@@ -0,0 +1,114 @@
+"""This Module contains implementations of State classes for Finite State Machines"""
+
+import abc
+import contextlib
+from typing import ClassVar, Optional
+from uuid import uuid4
+
+
+class State(abc.ABC):
+    __knows_uids: ClassVar[set[str]] = set()
+    __not_cache: ClassVar[dict[str, "_NOTState"]] = {}
+    __or_cache: ClassVar[dict[tuple[str, str], "_ORState"]] = {}
+    __and_cache: ClassVar[dict[tuple[str, str], "_ANDState"]] = {}
+    __xor_cache: ClassVar[dict[tuple[str, str], "_XORState"]] = {}
+
+    IDLE: "State"
+    """Default State for all Finite State Machines"""
+    ANY: "State"
+    """Special State that matches any other State. Useful to define fallback behavior.
+    *Not* supported in ``set_state`` method of FSMs.
+    """
+
+    def __init__(self, uid: Optional[str] = None):
+        effective_uid = uid or uuid4().hex
+        if effective_uid in self.__knows_uids:
+            raise ValueError(f"Duplicate UID: {effective_uid} already registered")
```

Comment:
is the error saying UID correct? I would understand the optional type hint to be anything hashable

Reply:
Sorry, I don't get you. What do you think is wrong? the phrasing of the error message?